sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/Constants.java

```diff
@@ -52,24 +52,105 @@ private Constants() {
      */
     public static final class ConnectionStringConstants {
         /**
-         * The AccountName key.
+         * The setting name for the storage account name.
          */
-        public static final String ACCOUNT_NAME = "accountname";
+        public static final String ACCOUNT_NAME_NAME = "AccountName";
 
         /**
-         * The AccountKey key.
+         * The setting name for the storage account key.
          */
-        public static final String ACCOUNT_KEY = "accountkey";
+        public static final String ACCOUNT_KEY_NAME = "AccountKey";
 
         /**
-         * The DefaultEndpointsProtocol key.
+         * The setting name for using the default storage endpoints with the specified protocol.
          */
-        public static final String ENDPOINT_PROTOCOL = "defaultendpointsprotocol";
+        public static final String DEFAULT_ENDPOINTS_PROTOCOL_NAME = "DefaultEndpointsProtocol";
 
         /**
-         * The EndpointSuffix key.
+         * The setting name for a custom blob storage endpoint.
          */
-        public static final String ENDPOINT_SUFFIX = "endpointsuffix";
+        public static final String BLOB_ENDPOINT_NAME = "BlobEndpoint";
+
+        /**
+         * The setting name for a custom blob storage secondary endpoint.
+         */
+        public static final String BLOB_SECONDARY_ENDPOINT_NAME = "BlobSecondaryEndpoint";
+
+        /**
+         * The setting name for a custom queue endpoint.
+         */
+        public static final String QUEUE_ENDPOINT_NAME = "QueueEndpoint";
+
+        /**
+         * The setting name for a custom queue secondary endpoint.
+         */
+        public static final String QUEUE_SECONDARY_ENDPOINT_NAME = "QueueSecondaryEndpoint";
```

Comment:
Can we generate secondary endpoint from portal? Or from CLI?

Reply:
I couldn't find a way to generate it from portal. It seems by appending `-secondary` to the blob name should result in secondary endpoint. I guess it's also possible to define a custom domain for secondary.